OpenRA.Mods.Common/Traits/Rearmable.cs

```diff
@@ -56,5 +121,43 @@ void INotifyResupply.BeforeResupply(Actor self, Actor target, ResupplyType types
 		}
 
 		void INotifyResupply.ResupplyTick(Actor self, Actor target, ResupplyType types) { }
+
+		string IOrderVoice.VoicePhraseForOrder(Actor self, Order order)
+		{
+			return order.OrderString == "Rearm" && CanRearm() ? Info.Voice : null;
+		}
+
+		void IResolveOrder.ResolveOrder(Actor self, Order order)
+		{
+			if (order.OrderString != "Rearm")
+				return;
+
+			// Repair orders are only valid for own/allied actors,
+			// which are guaranteed to never be frozen.
+			if (order.Target.Type != TargetType.Actor)
+				return;
+
+			// Aircraft handle Repair orders directly in the Aircraft trait
+			// TODO: Move the order handling of both this trait and Aircraft to a generalistic DockManager
+			if (isAircraft || isRepairable)
```

Comment:
Why not handle rearm-able only here? For backwards compability? All actors that currenty do not have the trait rearmable?

Reply:
Consider letting other traits call a method in this class that handle rearming. So that all rearm logic is in one place.

Reply:
I'm thinking of reworking this pr to add an abstract class that both `Rearmable` and `Repairable` inherit from, perhaps `Aircraft` too. It would contain the shared logic and be called something like `ReturnToBase` or as in the comments `DockManager`. Or perhaps a `ReturnToBase` trait instead?

 But I'm still not sure how best to handle collisions.